File: src/simulation/interactionmodel/InteractionModel.java
package simulation.interactionmodel;

import simulation.World;
import simulation.Event;

import java.util.Collection;

/**
 * Represents general model by which World changes.
 */
public interface InteractionModel {
    void turn(World world);

    void geologicTurn(World world);

    Collection<Event> getEvents();

    void clearEvents();
}

File: src/simulation/interactionmodel/InteractionModel.kt
package simulation.interactionmodel

import simulation.event.Event
import simulation.World

/**
 * Represents general model by which World changes.
 */
interface InteractionModel {
    fun turn(world: World)
    fun geologicTurn(world: World)
    val newEvents: List<Event>
    val allEvents: List<Event>
}